### PR TITLE
feat(digitsd): fix the voicemail message cap at 10 minutes, drop the setting

### PR DESCRIPTION
**Pull request by justinlindh**
## Summary

The per-line max-message-duration voicemail setting is removed. It was pointless configurability: a message ends when the caller hangs up, and the duration cap is only a backstop for a caller who never does. A fixed, generous cap covers that without a knob nobody needs to turn.

- `internal/voicemail/store.go`: new fixed package constant `messageMaxDuration` = 10 minutes; `Options.MaxMessageDuration` is removed and `BeginRecording` caps the recorder at the constant, mirroring how greeting recording uses `greetingMaxDuration`.
- `internal/config/config.go`: the `MaxMessageDuration` field is removed from the `Voicemail` config struct and `defaultVoicemail()`.
- `internal/signal/protocol.go`: `max_message_seconds` is dropped from the wire voicemail-settings struct, paired with the server-side removal.
- `cmd/digitsd/main.go`: the `line_settings` receiver no longer reads or applies a max-message value; the store opens with only `MaxStoredMessages`.
- `cmd/digitsd/voicemail.go`: at the 10-minute cap, the caller hears the prompt beep before the call is hung up, so a runaway recording is not dropped into silent dead air.
- `cmd/digitsd/voicemail.go`: **recovered guard.** The message-record loop now treats `ErrRecorderClosed` from `AppendFrame` as a clean, quiet case instead of logging a spurious WARN. It deliberately does NOT `return` (unlike the `*97` greeting loop) because this goroutine must keep decoding caller audio into `webrtcCh` for the live call after a pickup. This guard was reviewed and approved under PR #595 but lost when that PR squash-merged at its first-commit state; it is recovered here.
- Docs: `voicemail.md` and `voicemail-guide.md` updated for the fixed 10-minute cap and the cap beep; the configurable 15-180s language is removed.

Recording-ends-on-caller-hangup is unchanged.

## History note

This replaces PR #597, which was abandoned: a shared-worktree collision tangled go-backend's server commit into that branch. This branch is a clean digitsd-only commit off `main`. The server-side removal of `max_message_seconds` is in PR #596.

## Test plan

- [x] `go build ./...`, `go vet ./...`, `go test ./...`, `golangci-lint run ./...` all pass
- [ ] Hardware: leave a normal voicemail, hang up, confirm it saves and ends cleanly (unchanged path)
- [ ] Hardware: a recording that runs to the cap plays a beep to the caller before the call drops
- [ ] Hardware: record a `*97` greeting and verify no spurious `recorder closed` WARN

### CODE REVIEW

**Comment by justinlindh** [comment]
@claude

File: pi/digitsd/cmd/digitsd/voicemail.go [hunk 1]
```diff
@@ -362,9 +362,17 @@ func (d *daemonCallbacks) VoicemailAutoAnswer() {
...
 					if err != nil {
-						slog.Warn("voicemail: append frame failed", "caller", caller, "error", err)
+						// ErrRecorderClosed means VoicemailPickup finalized the
+						// recorder between the read above and this append. That
+						// is expected, not a failure, so stay quiet. Crucially
+						// do NOT return: unlike the *97 greeting loop this
+						// goroutine still has to decode caller audio into
+						// webrtcCh, which after pickup is the live call's audio.
+						if !errors.Is(err, voicemail.ErrRecorderClosed) {
+							slog.Warn("voicemail: append frame failed", "caller", caller, "error", err)
```
[the remark this answers is not in the document]
Comment (justinlindh) [reply pi/digitsd/cmd/digitsd/voicemail.go:372]: Thanks. That asymmetry (greeting loop returns, message loop does not) was the whole subtlety the recovered guard had to get right, hence the comment.

File: pi/digitsd/cmd/digitsd/voicemail.go [hunk 2]
```diff
@@ -373,6 +381,20 @@ func (d *daemonCallbacks) VoicemailAutoAnswer() {
...
+						// outbound beep slot, so the caller hears it. The sleep
+						// lets the beep drain before VoicemailRecordEnded tears
+						// the call down.
+						d.mu.Lock()
+						pipeline := d.pipeline
+						d.mu.Unlock()
+						if pipeline != nil {
+							pipeline.PlayGreetingBeep(500 * time.Millisecond)
+							time.Sleep(600 * time.Millisecond)
+						}
```
[the remark this answers is not in the document]
Comment (justinlindh) [reply pi/digitsd/cmd/digitsd/voicemail.go:397]: Skipping, with reason. This mirrors the established pattern in the same file: the VoicemailAutoAnswer greeting goroutine also captures `pipeline := d.pipeline` and then calls `PlayGreetingBeep` on it outside the lock. The pipeline is a GC-managed Go object, so a stale reference is still valid memory: there is no use-after-free or nil-deref. `PlayGreetingBeep` on a stopped pipeline is a harmless buffer write that nobody drains, not a panic. And this is a terminating path (the call is about to hang up). Holding d.mu across the call would be the inconsistent choice here. Leaving as-is.

File: pi/digitsd/internal/voicemail/store_test.go [hunk 3]
```diff
@@ -185,28 +185,34 @@ func TestFIFOEviction(t *testing.T) {
...
 		}
 	}
 	atCap, err := r.AppendFrame([]byte{0xff})
 	if err != nil {
 		t.Fatal(err)
 	}
 	if !atCap {
-		t.Errorf("expected atCap=true on 3rd frame")
+		t.Errorf("expected atCap=true on the cap-th frame (%d)", capFrames)
 	}
```
[the remark this answers is not in the document]
Comment (justinlindh) [reply pi/digitsd/internal/voicemail/store_test.go:216]: Skipping, with a measurement. I timed it: `go test -run TestMessageMaxDurationCap` completes in 11ms, not the estimated second or two (30000 small buffered writes to a temp file are cheap). It is not a bottleneck. The suggested split would need to either re-add a test-only cap-injection seam, which is exactly the configurable knob this PR removes, or duplicate the framesForDuration math in the test; neither is worth it for an 11ms test that genuinely exercises the real cap end to end. Leaving as-is.
